packages/cpp/src/browser/cpp-commands.ts

```diff
@@ -59,4 +99,30 @@ export class CppCommandContribution implements CommandContribution {
             });
         });
     }
+
+    protected changeConfiguration(): void {
+        this.workspaceService.tryRoot.then(async resolvedRoot => {
+            const root = resolvedRoot || await this.fileSystem.getCurrentUserHome();
+            const rootUri = new URI(root.uri).parent;
+            const rootStat = await this.fileSystem.getFileStat(rootUri.toString());
+            const rootNode = DirNode.createRoot(rootStat);
+            const dialog = this.fileDialogFactory({ title: Commands.OPEN.label! });
+            dialog.model.navigateTo(rootNode);
+            const node = await dialog.open();
+            let targetFolder: string = this.getTargetFolder(node);
+
+            this.clientContribution.languageClient.then((languageClient) => {
+                let configParams: ClangdConfigurationParams = { compilationDatabasePath: targetFolder };
+                let interfaceParams: DidChangeConfigurationParams = { settings: configParams };
+                languageClient.sendRequest(AnyRequest.type, interfaceParams);
+            });
+        });
+    }
+
+    protected getTargetFolder(node: Readonly<FileStatNode> | undefined): string {
+        if (!node) {
+            return "";
```

Comment:
That loooks wierd.. what happens with folder == empty string ?

Reply:
An assert will fail on server side because an empty string is not considered an absolute path. Should I just return / as the default or should I do something else?

Reply:
I think the server should be changed to log an error to the user but continue gracefully, i.e. just not find the compilation database and parsing might not be accurate.

Reply:
And the client could also not send the message at all if the folder is empty.